src/common/viz/FigureExtractor.js

```diff
@@ -17,210 +17,334 @@ define(['./Utils'], function (Utils) {
+                subgraphId: nodeInfo.parent_id,
+                lineName: nodeInfo.attributes.name,
+                label:  nodeInfo.attributes.label,
+                lineWidth: nodeInfo.attributes.lineWidth,
+                marker: nodeInfo.attributes.marker,
+                name: nodeInfo.attributes.name,
+                type: 'line',
+                points: points,
+                color: nodeInfo.attributes.color
+            };
+            return desc;
         }
-    };
 
-    FigureExtractor.prototype.getMetadataChildrenIds = function (node) {
-        const allMetaNodes = this._client.getAllMetaNodes();
-        const metadataBaseNode = allMetaNodes
-            .find(node => node.getAttribute('name') === BASE_METADATA_TYPE);
+        [EXTRACTORS.IMAGE] (nodeInfo) {
+            const id = nodeInfo.id,
+                execId = nodeInfo.execution_id,
+                imageHeight = nodeInfo.attributes.height,
+                imageWidth = nodeInfo.attributes.width,
+                numChannels = nodeInfo.attributes.numChannels;
+            const colorModel = numChannels === 3 ? 'rgb' : 'rgba';
+            return {
+                id: id,
+                execId: execId,
+                subgraphId: nodeInfo.parent_id,
+                type: 'image',
+                width: imageWidth,
+                height: imageHeight,
+                colorModel: colorModel,
+                visible: nodeInfo.attributes.visible,
+                rgbaMatrix: Utils.base64ToImageArray(nodeInfo.attributes.rgbaMatrix, imageWidth, imageHeight, numChannels)
+            };
+        }
+
+        [EXTRACTORS.SCATTER_POINTS] (nodeInfo) {
+            const id = nodeInfo.id,
+                execId = nodeInfo.execution_id;
+            let points, desc;
+
+            points = nodeInfo.attributes.points.split(';')
+                .filter(data => !!data)  // remove any ''
+                .map(pair => extractPointsArray(pair));
+            desc = {
+                id: id,
+                execId: execId,
+                subgraphId: nodeInfo.parent_id,
+                marker: nodeInfo.attributes.marker,
+                name: nodeInfo.attributes.name,
+                type: 'scatterPoints',
+                points: points,
+                width: nodeInfo.attributes.width,
+                color: nodeInfo.attributes.color
+            };
+
+            return desc;
+        }
+
+        getExecutionId (/* node */) {
+            throw new Error('getExecutionId is not implemented');
+        }
+
+        getGraphNode (/* node */) {
+            throw new Error('getGraphNode is not implemented');
+        }
+
+        _getContainmentParentNodeAt (/* node, metaType */) {
+            throw new Error('_getContainmentParentNodeAt is not implemented');
+        }
+
+        getMetaType (/* node */) {
+            throw new Error('getMetaType is not implmented');
+        }
+
+        GMENodeToMetadataJSON (/* node, shallow=false */) {
```

Comment:
I think this should be the only one that needs to be overridden. The others should accept the JSON representation instead.

Reply:
This also would reduce the risk of bugs since the tests only cover the case with the core. If the only difference was how they retrieved attributes, etc, to create the JSON representation, we would have better overall test coverage (w/o adding any more tests).